drivers/audio/wm8994.c

```diff
@@ -373,16 +450,9 @@ static void wm8994_writereg(FAR struct wm8994_dev_s *priv, uint16_t regaddr,
  *
  */
 
-static void wm8994_takesem(sem_t *sem)
+static int wm8994_takesem(sem_t *sem)
```

Comment:
```suggestion
static int wm8994_takesem(FAR sem_t *sem)
```

Reply:
Short question. Wasn't the FAR to be removed from the code?

Reply:
The `FAR` is removed for archs that do not support far and near segment addressing like ARM, RISC-V, Xtensa, etc.
So the answer is: `FAR` should be added for common code and should not be added for archs that do not support it.